docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+
+``` json
+[{"n":"urn:dev:ow:10e2073a01080063","u":"Cel","v":23.1}]
+```
+
+In the example above, the array (what EdgeX would consider a collection of readings) has a single SenML Record with a
+measurement for a sensor named "urn:dev:ow:10e2073a01080063" with a current value of 23.1 for degrees measured in Celsius (Cel) unit of measure.  However, SenML suggests the use of short names for the keys in most cases, but long names could be used.  In which case, the JSON SenML reading would look like the following:
+
+``` json
+[{"Name":"urn:dev:ow:10e2073a01080063","Unit":"Cel","Value":23.1}]
+```
+
+In this way, the parallels to EdgeX model are, by accident, uncanny - at least in the JSON instance.  SenML goes to much more depth to provide extensions and more definitions around measurements.  But at its base, the EdgeX format is not unlike SenML and could easily be aligned with SenML in the future (or allow for an application service to export in SenML with an additional function fairly easily and if there were demand).
+
+However, on the basis of "unit of measure", SenML is actually light on details.  This ADR and the proposal here goes much further than what SenML specifies.  With regard to UoM, the SenML specification only says:
+
+!!! Quote
+    If the Record has no Unit, the Base Unit is used as the Unit.  Having no Unit and no Base Unit is allowed; any information that may be required about units applicable to the value then needs to be provided by the application context.
+
+Therefore, SenML should be examined for future versions of EdgeX with regard to data model, but its relevance to **unit of measure** is believed to be minimal at this time.
+
+### Future Considerations/Additions/Improvements
+
+In the future, validation may be turned `on` or `off` per device service; allowing the decision to validate units of measure to be accomplished on a service or even allow the device service to validate/not validate based on particular devices.
+
+In the future, additional criteria may be added to the unit of measure information to all for more specific (or allowing more granularity) validation.  For example, the category of units of measure could be specified in a device resource so that a profile's units are validated against specific sources or collections of unit of measure.
+
+Use of URI to specify the unit of measures, when using external UoM definitions, is novel with regard to providing EdgeX information.  Per core working group of 2/17/22, we are looking to use a URI to specify an external configuration file to specify UoM.  If this is successful, this URI mechanism may be used to point to device profiles, configuration files, and other information in the future.  This would even allow multiple EdgeX instances to use the same configuration or profile (multiple EdgeX instances using the same URI to use a shared profile for example).
```

Comment:
Unless there were public meeting notes for the WG meeting on 2/17/22, I'm not sure it adds value to mention that a decision that impacts this ADR was made in that meeting? What do others think?

Reply:
We wanted to make sure it was captured for consideration.  It was also added to the [EdgeX 3.0 board](https://github.com/orgs/edgexfoundry/projects/48).  Once we get approval of this ADR and all considerations / conversations are taken care of, I can remove it from this ADR.